pkg/endpoint/bpf.go

```diff
@@ -1071,6 +1074,27 @@ func (e *Endpoint) SkipStateClean() {
 	e.unlock()
 }
 
+func (e *Endpoint) initPolicyMapPressureMetric() {
+	if !option.Config.MetricsConfig.BPFMapPressure {
+		return
+	}
+
+	if e.policyMapPressureGauge != nil {
+		return
+	}
+
+	e.policyMapPressureGauge = metrics.NewBPFMapPressureGauge(e.policyMap.NonPrefixedName(), policymap.PressureMetricThreshold)
+}
+
+func (e *Endpoint) updatePolicyMapPressureMetric() {
+	if e.policyMapPressureGauge == nil {
+		return
+	}
+
+	value := float64(len(e.realizedPolicy.PolicyMapState)) / float64(e.policyMap.MapInfo.MaxEntries)
+	e.policyMapPressureGauge.Set(value)
+}
```

Comment:
One of the reasons I was drawn to look at [0] was that I noticed this function appears to be equivalent but different in implementation due to the code I pointed out at [0]. Ideally these should be consistent (Ideally using the same object definition, shared across both users; but there's only a few lines so that's not such a big deal).

Would you mind either consistently doing the config check in the init() or in the update() across both implementations?

Reply:
Here is different because we can check `option.Config.MetricsConfig.BPFMapPressure` on initialization which saves doing extra checks in the update.